### ISSUE TITLE
expo with turborepo not working

### ISSUE THREAD

**Issue by princejoogie** [expo/expo#18693]
### Summary

I'm trying to add a **FRESH** `(yarn create expo-app)` expo app in my existing monorepo (with turborepo) that has a web app and some packages for prisma and graphql. I followed the official guide [here](https://docs.expo.dev/guides/monorepos/) but am getting the error as seen in the image below.

![image](https://user-images.githubusercontent.com/47204120/185339532-aae4516d-9e77-4a8d-80c1-bb3a197403fa.png)

![image](https://user-images.githubusercontent.com/47204120/185341752-2090c4cb-9fb5-412a-b177-d6579ca6ef1b.png)

**Additional Context**:
- didnt change any of the code from `App.tsx` when first created from `yarn create expo-app`
- app was working prior to moving it into the monorepo

### What platform(s) does this occur on?

Android, iOS, Web

### SDK Version

46.0.7

### Environment

```sh
expo-env-info 1.0.5 environment info:
  System:
    OS: Linux 5.10 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
    Shell: 5.8.1 - /usr/bin/zsh
  Binaries:
    Node: 16.16.0 - ~/.config/nvm/versions/node/v16.16.0/bin/node
    Yarn: 1.22.19 - ~/.config/nvm/versions/node/v16.16.0/bin/yarn
    npm: 8.11.0 - ~/.config/nvm/versions/node/v16.16.0/bin/npm
  SDKs:
    Android SDK:
      API Levels: 33
      Build Tools: 33.0.0
  npmGlobalPackages:
    eas-cli: 0.60.0
    expo-cli: 6.0.2
  Expo Workflow: managed
```

### Minimal reproducible example

https://github.com/princejoogie/expo-issue-reproduction

**Comment by princejoogie** [comment]
closing this issue as it is fixed.

**TL;DR**
use the same version of `react` across apps in a monorepo
